Bitbucket raw url not working

We host our own Bitbucket server and when I try to execute jbang with the raw link to the file, I always get an error.  I am sorry but I can not give you a working link.

```
user@home:~$ jbang --verbose https://bitbucket.foobar.net/projects/JBAN/repos/examples/raw/src/main/java/examples/jfxtiles.java
[jbang] Downloaded file https://bitbucket.foobar.net/projects/JBAN/repos/examples/raw/src/main/java/examples/jfxtiles.java
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] Building jar...
[jbang] compile: /opt/java/sdk/graalvm/bin/javac -classpath /home/user/.m2/repository/eu/hansolo/tilesfx/11.30/tilesfx-11.30.jar:/home/user/.m2/repository/com/googlecode/json-simple/json-simple/1.1.1/json-simple-1.1.1.jar:/home/user/.m2/repository/junit/junit/4.10/junit-4.10.jar:/home/user/.m2/repository/org/openjfx/javafx-media/12.0.1/javafx-media-12.0.1.jar:/home/user/.m2/repository/org/openjfx/javafx-media/12.0.1/javafx-media-12.0.1-linux.jar:/home/user/.m2/repository/org/openjfx/javafx-controls/12.0.1/javafx-controls-12.0.1.jar:/home/user/.m2/repository/org/openjfx/javafx-controls/12.0.1/javafx-controls-12.0.1-linux.jar:/home/user/.m2/repository/org/openjfx/javafx-web/12.0.1/javafx-web-12.0.1.jar:/home/user/.m2/repository/org/openjfx/javafx-web/12.0.1/javafx-web-12.0.1-linux.jar:/home/user/.m2/repository/org/openjfx/javafx-base/12.0.1/javafx-base-12.0.1.jar:/home/user/.m2/repository/org/openjfx/javafx-base/12.0.1/javafx-base-12.0.1-linux.jar:/home/user/.m2/repository/org/openjfx/javafx-graphics/12.0.1/javafx-graphics-12.0.1.jar:/home/user/.m2/repository/org/openjfx/javafx-graphics/12.0.1/javafx-graphics-12.0.1-linux.jar -d /home/user/.jbang/cache/jars/"jfxtiles.java"; filename*=UTF-8''jfxtiles.java.a559f83ce53db9459dc66a853ea4d72d27050beb88a0d13ec0b70a84cd2ac5cd /home/user/.jbang/cache/urls/9d201cddd21d493a35432a75483dc1e1f3629a1eb18f3d536e7a09f30e7e8eaf/"jfxtiles.java"; filename*=UTF-8''jfxtiles.java
/home/user/.jbang/cache/urls/9d201cddd21d493a35432a75483dc1e1f3629a1eb18f3d536e7a09f30e7e8eaf/"jfxtiles.java"; filename*=UTF-8''jfxtiles.java:82: error: class jfxtiles is public, should be declared in a file named jfxtiles.java
public class jfxtiles extends Application {
       ^
Note: /home/user/.jbang/cache/urls/9d201cddd21d493a35432a75483dc1e1f3629a1eb18f3d536e7a09f30e7e8eaf/"jfxtiles.java"; filename*=UTF-8''jfxtiles.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error
[jbang] Deleting folder /home/user/.jbang/cache/jars/"jfxtiles.java"; filename*=UTF-8''jfxtiles.java.a559f83ce53db9459dc66a853ea4d72d27050beb88a0d13ec0b70a84cd2ac5cd
[jbang] [ERROR] Error during compile
dev.jbang.ExitException: Error during compile
        at dev.jbang.cli.BaseBuildCommand.buildJar(BaseBuildCommand.java:210)
        at dev.jbang.cli.BaseBuildCommand.build(BaseBuildCommand.java:137)
        at dev.jbang.cli.Run.prepareArtifacts(Run.java:83)
        at dev.jbang.cli.Run.doCall(Run.java:71)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:85)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:12)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at dev.jbang.Main.main(Main.java:14)
[jbang] If you believe this a bug in jbang open issue at https://github.com/jbangdev/jbang/issues
```
```
wget https://bitbucket.foobar.net/projects/JBAN/repos/examples/raw/src/main/java/examples/jfxtiles.java
jbang jfxtiles.java
```

no errors

`
jbang https://raw.githubusercontent.com/jbangdev/jbang-examples/master/examples/jfxtiles.java`

no errors

--- a/src/main/java/dev/jbang/Util.java
+++ b/src/main/java/dev/jbang/Util.java
@@ -6,11 +6,13 @@ import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
 import java.net.HttpURLConnection;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
 import java.net.URLConnection;
+import java.net.URLDecoder;
 import java.nio.channels.Channels;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.charset.StandardCharsets;
@@ -478,14 +480,7 @@ public class Util {
 
 				if (disposition != null) {
 					// extracts file name from header field
-					int index = disposition.indexOf("filename=");
-					if (index > 0) {
-						fileName = disposition.substring(index + 9);
-						// Seems not everybody properly quotes the filename
-						if (fileName.startsWith("\"") && fileName.endsWith("\"")) {
-							fileName = fileName.substring(1, fileName.length() - 1);
-						}
-					}
+					fileName = getDispositionFilename(disposition);
 				}
 
 				if (fileName.trim().isEmpty()) {
@@ -522,6 +517,37 @@ public class Util {
 
 		return saveFilePath;
 
+	}
+
+	public static String getDispositionFilename(String disposition) {
+		String fileName = "";
+		int index = disposition.indexOf("filename=");
+		if (index > 0) {
+			fileName = unquote(disposition.substring(index + 9));
+			// Seems not everybody properly quotes the filename
+			fileName = unquote(fileName);
+		} else {
+			index = disposition.indexOf("filename*=");
+			if (index > 0) {
+				String encodedName = unquote(disposition.substring(index + 10));
+				String[] parts = encodedName.split("'", 3);
+				if (parts.length == 3) {
+					try {
+						fileName = URLDecoder.decode(parts[2], parts[0]);
+					} catch (UnsupportedEncodingException e) {
+						Util.infoMsg("Content-Disposition contains unsupported encoding " + parts[0]);
+					}
+				}
+			}
+		}
+		return fileName;
+	}
+
+	public static String unquote(String txt) {
+		if (txt.startsWith("\"") && txt.endsWith("\"")) {
+			txt = txt.substring(1, txt.length() - 1);
+		}
+		return txt;
 	}
 
 	/**

--- a/src/test/java/dev/jbang/TestUtil.java
+++ b/src/test/java/dev/jbang/TestUtil.java
@@ -2,6 +2,7 @@ package dev.jbang;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasItem;
 import static org.hamcrest.Matchers.not;
 import static org.junit.jupiter.api.Assertions.assertFalse;
@@ -87,5 +88,12 @@ public class TestUtil extends BaseTest {
 		assertThat(p, containsInAnyOrder("res/resource.java"));
 		assertThat(p, not(hasItem("test.java")));
 
+	}
+
+	@Test
+	void testDispostionFilename() throws IOException {
+		assertThat(Util.getDispositionFilename("inline; filename*=iso-8859-1'en'%A3%20rates"), equalTo("£ rates"));
+		assertThat(Util.getDispositionFilename("inline; filename*=\"UTF-8''%c2%a3%20and%20%e2%82%ac%20rates\""),
+				equalTo("£ and € rates"));
 	}
 }

